travis-infrastructure/terraform-config
# Increase BOOT_PRE_POLL_SLEEP for workers in Production 5 to 25s

## What is the problem that this PR is trying to fix?
Precise builds are being requeued a lot and it seems to be caused by the worker attempting to ssh into the instance too soon. 

## What approach did you choose and why?
This is an attempt to remediate things without adding any invasive changes so close to the holidays. I'd like to see if this improves the requeue rate in Production 5.

## How can you test this?
This is the output of `make plan` (with the user-data removed for better visibility):

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place
-/+ destroy and then create replacement

Terraform will perform the following actions:

-/+ module.gce_project_1.google_compute_firewall.deny_target_ip (new resource required)
      id:                            "deny-target-ip" => <computed> (forces new resource)
      deny.#:                        "3" => "3"
      deny.1367131964.ports.#:       "0" => "0"
      deny.1367131964.protocol:      "icmp" => "icmp"
      deny.1486604749.ports.#:       "0" => "0"
      deny.1486604749.protocol:      "udp" => "udp"
      deny.3848845357.ports.#:       "0" => "0"
      deny.3848845357.protocol:      "tcp" => "tcp"
      destination_ranges.#:          "1" => "1"
      destination_ranges.3294923498: "194.116.96.25/32" => "194.116.96.25/32"
      direction:                     "EGRESS" => "EGRESS"
      name:                          "deny-target-ip" => "deny-target-ip"
      network:                       "https://www.googleapis.com/compute/v1/projects/travis-ci-prod-5/global/networks/main" => "main"
      priority:                      "1000" => "0" (forces new resource)
      project:                       "travis-ci-prod-5" => "travis-ci-prod-5"
      self_link:                     "https://www.googleapis.com/compute/v1/projects/travis-ci-prod-5/global/firewalls/deny-target-ip" => <computed>
      source_ranges.#:               "0" => <computed>

  ~ module.gce_project_1.google_compute_instance.bastion-b
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

  ~ module.gce_project_1.module.gce_worker_a.google_compute_instance.worker_org[0]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

  ~ module.gce_project_1.module.gce_worker_a.google_compute_instance.worker_org[1]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

-/+ module.gce_project_1.module.gce_worker_a.null_resource.worker_com_validation (new resource required)
      id:                            "2543483969303764883" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "c5f24072027587b30c81a94d625b9a5d13e1c1648356dc922041bb8d63ce82b0" => "b9eaf49d5ddd31793cacaab93f61bc227a53e9ff1dc7fb1f908665b9442390b3" (forces new resource)

-/+ module.gce_project_1.module.gce_worker_a.null_resource.worker_org_validation (new resource required)
      id:                            "2888779851703429759" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "36e3ea171c208a42763c60d3cd7ed3689d279af5cfebf4d5ee7cae222c3cbb19" => "8e8f06f163376465ce646649f3cf5f80fd8c30c7e8a79c70677ccd43ccdd2729" (forces new resource)

  ~ module.gce_project_1.module.gce_worker_b.google_compute_instance.worker_org[0]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

  ~ module.gce_project_1.module.gce_worker_b.google_compute_instance.worker_org[1]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

-/+ module.gce_project_1.module.gce_worker_b.null_resource.worker_com_validation (new resource required)
      id:                            "1629174679522947456" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "c5f24072027587b30c81a94d625b9a5d13e1c1648356dc922041bb8d63ce82b0" => "b9eaf49d5ddd31793cacaab93f61bc227a53e9ff1dc7fb1f908665b9442390b3" (forces new resource)

-/+ module.gce_project_1.module.gce_worker_b.null_resource.worker_org_validation (new resource required)
      id:                            "5875064014195153601" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "36e3ea171c208a42763c60d3cd7ed3689d279af5cfebf4d5ee7cae222c3cbb19" => "8e8f06f163376465ce646649f3cf5f80fd8c30c7e8a79c70677ccd43ccdd2729" (forces new resource)

  ~ module.gce_project_1.module.gce_worker_c.google_compute_instance.worker_org[0]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

  ~ module.gce_project_1.module.gce_worker_c.google_compute_instance.worker_org[1]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

-/+ module.gce_project_1.module.gce_worker_c.null_resource.worker_com_validation (new resource required)
      id:                            "2365087644011957208" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "c5f24072027587b30c81a94d625b9a5d13e1c1648356dc922041bb8d63ce82b0" => "b9eaf49d5ddd31793cacaab93f61bc227a53e9ff1dc7fb1f908665b9442390b3" (forces new resource)

-/+ module.gce_project_1.module.gce_worker_c.null_resource.worker_org_validation (new resource required)
      id:                            "9026233376377978819" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "36e3ea171c208a42763c60d3cd7ed3689d279af5cfebf4d5ee7cae222c3cbb19" => "8e8f06f163376465ce646649f3cf5f80fd8c30c7e8a79c70677ccd43ccdd2729" (forces new resource)

  ~ module.gce_project_1.module.gce_worker_f.google_compute_instance.worker_org[0]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...

  ~ module.gce_project_1.module.gce_worker_f.google_compute_instance.worker_org[1]
      metadata.user-data:            "#cloud-config\n# vim:filetype=yaml ...
      
-/+ module.gce_project_1.module.gce_worker_f.null_resource.worker_com_validation (new resource required)
      id:                            "7841520030256671828" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "c5f24072027587b30c81a94d625b9a5d13e1c1648356dc922041bb8d63ce82b0" => "b9eaf49d5ddd31793cacaab93f61bc227a53e9ff1dc7fb1f908665b9442390b3" (forces new resource)

-/+ module.gce_project_1.module.gce_worker_f.null_resource.worker_org_validation (new resource required)
      id:                            "4455472707530178115" => <computed> (forces new resource)
      triggers.%:                    "1" => "1"
      triggers.config_signature:     "36e3ea171c208a42763c60d3cd7ed3689d279af5cfebf4d5ee7cae222c3cbb19" => "8e8f06f163376465ce646649f3cf5f80fd8c30c7e8a79c70677ccd43ccdd2729" (forces new resource)


Plan: 9 to add, 9 to change, 9 to destroy.
```


## What feedback would you like, if any?

Does that plan make sense? I was a bit surprised by the extent of the changes, but I get the same results (with slightly different user-data, which is expected) if I run `make plan` on master as well.
